conans/model/info.py

```diff
@@ -481,4 +486,35 @@ def shared_library_package_id(self):
             for dep_name in self.requires.pkg_names:
                 dep_options = self.full_options[dep_name]
                 if "shared" not in dep_options or not self.full_options[dep_name].shared:
-                    self.requires[dep_name].package_revision_mode()
\ No newline at end of file
+                    self.requires[dep_name].package_revision_mode()
+
+    def parent_compatible(self, *_, **kwargs):
+        """If a built package for Intel has to be compatible for a Visual/GCC compiler
+        (consumer). Transform the visual/gcc full_settings into an intel one"""
+
+        if "compiler" not in kwargs:
+            raise ConanException("Specify 'compiler' as a keywork argument. e.g: "
+                                 "'parent_compiler(compiler=\"intel\")' ")
+
+        self.settings.compiler = kwargs["compiler"]
+        # You have to use here a specific version or create more than one version of
+        # compatible packages
+        kwargs.pop("compiler")
+        for setting_name in kwargs:
+            # Won't fail even if the setting is not valid, there is no validation at info
+            setattr(self.settings.compiler, setting_name, kwargs[setting_name])
+        self.settings.compiler.base = self.full_settings.compiler
+        for field in self.full_settings.compiler.fields:
+            value = getattr(self.full_settings.compiler, field)
+            setattr(self.settings.compiler.base, field, value)
+
+    def base_compatible(self):
+        """If a built package for Visual/GCC has to be compatible for an Intel compiler
+          (consumer). Transform the Intel profile into an visual/gcc one"""
+        if not self.full_settings.compiler.base:
```

Comment:
I'd say to remove this check, or raise an exception. If ``package_id()`` says that it is base_compatible(), the compiler should have defined a ``base`` field.

Reply:
Yes, I forgot. Will add a test.